compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -314,7 +314,7 @@ public BType checkType(Location pos,
     public boolean isLaxFieldAccessAllowed(BType type) {
         type = Types.getImpliedType(type);
         Set<BType> visited = new HashSet<>();
-        return isLaxType(type, visited) == 1 || type.tag == TypeTags.XML || type.tag == TypeTags.XML_ELEMENT;
+        return isLaxType(type, visited) == 1 || isAssignable(type, symTable.xmlType);
```

Comment:
In the previous implementation, this was also incorrect. Lax field access should only be allowed for lax types. However, XML is not a lax type. I have concerns about including this logic within this function.

Reply:
Similar reason to the above comment, as per the offline discussion